src/query/interpreter.cpp

```diff
@@ -1909,6 +1922,142 @@ PreparedQuery PrepareSettingQuery(ParsedQuery parsed_query, const bool in_explic
+                                     const std::optional<std::string> &username, const Parameters &parameters,
+                                     InterpreterContext *interpreter_context, DbAccessor *db_accessor) {
+  Frame frame(0);
+  SymbolTable symbol_table;
+  EvaluationContext evaluation_context;
+  evaluation_context.timestamp = QueryTimestamp();
+  evaluation_context.parameters = parameters;
+  ExpressionEvaluator evaluator(&frame, symbol_table, evaluation_context, db_accessor, storage::View::OLD);
+
+  bool isAdmin = interpreter_context->auth_checker->IsUserAdmin(username);
+
+  Callback callback;
+  switch (transaction_query->action_) {
+    case TransactionQueueQuery::Action::SHOW_TRANSACTIONS: {
+      callback.header = {"username", "transaction_id", "query"};
+      callback.fn = [handler = TransactionQueueQueryHandler(), interpreter_context, username, isAdmin]() mutable {
+        return handler.ShowTransactions(interpreter_context, username, isAdmin);
+      };
+      break;
+    }
+    case TransactionQueueQuery::Action::TERMINATE_TRANSACTIONS: {
+      std::vector<std::string> maybe_kill_transaction_ids;
+      std::transform(transaction_query->transaction_id_list_.begin(), transaction_query->transaction_id_list_.end(),
+                     std::back_inserter(maybe_kill_transaction_ids), [&evaluator](Expression *expression) {
+                       return std::string(expression->Accept(evaluator).ValueString());
+                     });
+      callback.header = {"transaction_id", "killed"};
+      callback.fn = [handler = TransactionQueueQueryHandler(), interpreter_context, maybe_kill_transaction_ids,
+                     username, isAdmin]() mutable {
+        return handler.KillTransactions(interpreter_context, maybe_kill_transaction_ids, username, isAdmin);
+      };
+      break;
+    }
+  }
+
+  return callback;
+}
+
+PreparedQuery PrepareTransactionQueueQuery(ParsedQuery parsed_query, const std::optional<std::string> &username,
+                                           const bool in_explicit_transaction, InterpreterContext *interpreter_context,
+                                           DbAccessor *dba) {
+  if (in_explicit_transaction) {
+    throw VersionInfoInMulticommandTxException();
+  }
+
+  auto *transaction_queue_query = utils::Downcast<TransactionQueueQuery>(parsed_query.query);
+  MG_ASSERT(transaction_queue_query);
+  auto callback =
+      HandleTransactionQueueQuery(transaction_queue_query, username, parsed_query.parameters, interpreter_context, dba);
+
+  return PreparedQuery{std::move(callback.header), std::move(parsed_query.required_privileges),
+                       [callback_fn = std::move(callback.fn), pull_plan = std::shared_ptr<PullPlanVector>{nullptr}](
+                           AnyStream *stream, std::optional<int> n) mutable -> std::optional<QueryHandlerResult> {
+                         if (UNLIKELY(!pull_plan)) {
```

Comment:
instead of `UNLIKELY` you can use [[unlikely]] from c++ 20, as we use in `PrepareShowConfigQuery`

Reply:
@as51340 maybe you know, but why is this path unlikely to execute. If I understand correctly, UNLIKELY means that this statement is unlikely to execute, due to the fact that there will be pull_plan. But I am not sure where do we set pull plan to be something else than `nullptr`. Do you know that part maybe?

Reply:
@as51340 maybe you know, but why is this path unlikely to execute. If I understand correctly, UNLIKELY means that this statement is unlikely to execute, due to the fact that there will be pull_plan. But I am not sure where do we set pull plan to be something else than `nullptr`. Do you know that part maybe?

Reply:
@antoniofilipovic I don't know I reused this part of the code. @antaljanosbenjamin please help.